### PR TITLE
File System Representation fixes

### PR DESCRIPTION
**Pull request by compnerd**

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @millenomi @phausler @parkera

**Comment by compnerd** [comment]
@swift-ci please test

File: CoreFoundation/URL.subproj/CFURL.c [hunk 1]
```diff
@@ -4459,7 +4459,23 @@ CFStringRef CFURLCreateStringWithFileSystemPath(CFAllocatorRef allocator, CFURLR
...
+          break;
+        case kCFURLHFSPathStyle:
+          relPathIsRelative = length > 0 && CFStringGetCharacterAtIndex(relPath, 0) == ':';
+          break;
+        case kCFURLWindowsPathStyle:
+          // This is not really ideal, but, since a drive letter a single
+          // character, and ':' is not a valid path character, and ':' is
+          // reserved for the ADS separator, this works.  In theory, on
+          // TARGET_DEPLOYMENT_WINDOWS we could use PathIsRelativeW here.
+          relPathIsRelative = length > 1 && CFStringGetCharacterAtIndex(relPath, 1) != ':';
```
Comment (millenomi) [inline CoreFoundation/URL.subproj/CFURL.c]: Per discussion: this needs to cover UNC paths, and then it’s good to go.

Comment (compnerd) [reply CoreFoundation/URL.subproj/CFURL.c]: Added a check for the UNC paths too.

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Pull request merged**